### PR TITLE
ceph: Mon failover should first scale down the failed mon before starting the new mon

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
When a mon is failed over, the new mon is started before the old mon is removed from quorum and purged from the cluster. If the mon has anti-affinity, in small clusters or specific topologies where sufficient nodes are not available, the new mon may be prevented from starting simply because the dead mon pod still exists. To ensure a clean failover, the dead mon pod must first be stopped before the new mon is started.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

// known CI issue
[skip ci]

### CODE REVIEW

File: pkg/operator/ceph/cluster/mon/health.go [hunk 1]
```diff
@@ -247,9 +247,50 @@ func (c *Cluster) failMon(monCount, desiredMonCount int, name string) {
 	}
 }
 
+func (c *Cluster) enableMonDaemon(name string, enabled bool) error {
```
Comment (leseb) [inline pkg/operator/ceph/cluster/mon/health.go:250]: I'm confused by the name of this method. Should it be something like `updateMonDepReplica`?

File: pkg/operator/ceph/cluster/mon/health.go [hunk 2]
```diff
@@ -247,9 +247,50 @@ func (c *Cluster) failMon(monCount, desiredMonCount int, name string) {
...
+		logger.Warningf("failed to stop mon %q for failover. %v", name, err)
+	}
+	newMonSucceeded := false
+	defer func() {
+		if newMonSucceeded {
+			// do nothing if the new mon was started successfully
+			return
+		}
+		if err := c.enableMonDaemon(name, true); err != nil {
+			// attempt to continue with the failover even if the bad mon could not be stopped
```
Comment (leseb) [inline pkg/operator/ceph/cluster/mon/health.go:289]: Is this really true? We could exit that method on L308 and this wouldn't mean that the bad mon could not be stopped right?

File: pkg/operator/ceph/cluster/mon/health.go (hunk 1 above)
Comment (travisn) [reply pkg/operator/ceph/cluster/mon/health.go:250]: Yes, I'll rename it for clarity

File: pkg/operator/ceph/cluster/mon/health.go (hunk 2 above)
Comment (travisn) [reply pkg/operator/ceph/cluster/mon/health.go:289]: This comment is misleading. Is this better?
```suggestion
			// attempt to continue even if the bad mon could not be restarted
```

**Pull request merged**